### PR TITLE
GH-2302: Clarify in Docs the DSL log() behavior

### PR DESCRIPTION
**Pull request by artembilan**
Fixes: spring-projects/spring-integration/issues/2302

### CODE REVIEW

File: src/reference/asciidoc/dsl.adoc [hunk 1]
```diff
@@ -491,6 +491,34 @@ public IntegrationFlow loggingFlow() {
 }
 ----
 
+[IMPORTANT]
+=====
+The `log()` or `wireTap()` opearators are applied to the current `MessageChannel` (if it is an instane of `ChannelInterceptorAware`) or an intermediate `DirectChannel` is injected into the flow for the currently configured endpoint.
```
Comment (garyrussell) [inline src/reference/asciidoc/dsl.adoc]: `s/instane/instance/`

File: src/reference/asciidoc/dsl.adoc [hunk 2]
```diff
@@ -491,6 +491,34 @@ public IntegrationFlow loggingFlow() {
...
+In the example above the `WireTap` interceptor is added to the `myChannel` directly, because `DirectChannel` implements `ChannelInterceptorAware`:
```
Comment (garyrussell) [inline src/reference/asciidoc/dsl.adoc]: `s/DirectChannel/QueueChannel/`

File: src/reference/asciidoc/dsl.adoc [hunk 3]
```diff
@@ -491,6 +491,34 @@ public IntegrationFlow loggingFlow() {
...
+
+When current `MessageChannel` doesn't implement `ChannelInterceptorAware`, an implicit `DirectChannel` and `BridgeHandler` are injected into the `IntegrationFlow` and only after that a `WireTap` is added to this new `DirectChannel`.
+And when there is no any channel declaration like in this sample:
+[source,java]
+----
+.handle(...)
+.log()
+}
+----
+=====
```
Comment (garyrussell) [inline src/reference/asciidoc/dsl.adoc]: You can't put `[source]` within an `IMPORTANT` block - see here https://docs.spring.io/spring-integration/docs/5.0.0.BUILD-SNAPSHOT/reference/html/java-dsl.html#java-dsl-wiretap

**Comment by garyrussell** [comment]
Merged as 75b5948c47804bb1703b4fa1a8f1f3697a5319ff

File: src/reference/asciidoc/dsl.adoc (hunk 2 above)
Comment (artembilan) [reply src/reference/asciidoc/dsl.adoc]: Why? `myChannel()` is a `DirectChannel`, not `QueueChannel`.

Maybe s/above/below? 😄 

Sorry for my bad English

**Comment by artembilan** [comment]
Merged by @garyrussell .

See comment above.

File: src/reference/asciidoc/dsl.adoc (hunk 2 above)
Comment (artembilan) [reply src/reference/asciidoc/dsl.adoc]: Fixed via: https://github.com/spring-projects/spring-integration/commit/2f450f7cde1df2c1596e4a00664e01295f317f3e

Comment (garyrussell) [reply src/reference/asciidoc/dsl.adoc]: hehe

**Pull request closed without merging**